ci: update yarn to v1.13.0

devversion opened the pull request:
Currently our version of Yarn is installed through the "circleci/node"
docker image. This is problematic because in order to be able to update
Yarn, we always need to update the docker image to a version that comes
with the desired Yarn version. Sometimes there is no docker image with the
desired latest Yarn version, and therefore we cannot easily update the Yarn version.

Additionally updating the docker image also means that we need to update our
version of NodeJS, as well as the version of `openssl` might have changed
(meaning that our encrypted credential files may not be decodable with the new
version of `openssl`)

IgorMinar commented on .circleci/config.yml:
```diff
@@ -59,6 +59,15 @@ var_9: &setup_circleci_bazel_config
+var_10: &download_yarn
+  run:
+    name: Downloading Yarn
+    command: curl -o- -L https://yarnpkg.com/install.sh | PROFILE=$BASH_ENV bash -s -- --version "1.13.0"
```
can you move the version number to the define_env_vars script please? we should keep all the version numbers in one place.

IgorMinar reviewed (approved):
lgtm with one small change requested. thanks

IgorMinar commented:
oh.. there is more changes we need to do.. sec.

IgorMinar commented:
@devversion 
- can you also please reclassify this change to "ci: "
- can you also bump the yarn version in package.json to require this version and drop old versions? thanks

IgorMinar commented:
- aio/package.json
- package.json

gkalpak commented on .circleci/config.yml:
```diff
@@ -59,6 +59,15 @@ var_9: &setup_circleci_bazel_config
+# docker image because this means that we can only use the Yarn version that comes with
+# a specific version of NodeJS. We want to be able to update Yarn without having to
+# update the docker image that comes with NodeJS.
+var_10: &download_yarn
```
Since we always need to download yarn before running `yarn install`, would it make sense to combine the two?

devversion replied:
There are a few jobs like `test_aio` that don't require the root `node_modules`, so for these jobs we need a way to _only_ download Yarn. do I miss something?

IgorMinar replied:
how about combining it with define_env_vars and renaming that to "env_init" or something along those lines?

devversion replied:
That sounds like a good idea. I can do a follow-up PR that combines these two YAML anchors into a `*init_environment` anchor (or whatever we call it)

devversion committed:
build: update yarn to v1.13.0

Currently our version of Yarn is installed through
the "circleci/node" docker image. This is problematic
because in order to be able to update Yarn, we always
need to update the docker image to a version that
comes with the desired Yarn version. Sometimes there
is no docker image with the desired latest Yarn version,
and therefore we cannot easily update the Yarn version.

Additionally updating the docker image also means that
we need to update our version of NodeJS, as well as the
version of `openssl` might have changed (meaning that
our encrypted credential files may not be decodable with
the new version of `openssl`)

IgorMinar commented:
btw we screwed up a bit here, this should have been master&patch PR.

general rule: all infra changes should be cherrypicked to patch branch whenever possible to minimize git conflicts when we need to cherry pick important infra fixes in the future.

I'll cp this PR into 7.2.x

gkalpak commented:
For context:
This was `master & patch`, but @matias [mentioned][1] there were conflicts with patch, so @devversion created a new PR for patch.

But, yes, we should try to keep the CI config consistent between master and patch, so that there won't be conflicts.

[1]: https://angular-team.slack.com/archives/C07DT5M6V/p1549403391556900
